src/components/CampaignTextersForm.jsx

```diff
@@ -168,16 +171,17 @@ export default class CampaignTextersForm extends React.Component {
         newTexter.assignment.contactsCount = texter.assignment.needsMessageCount + messagedCount
         return newTexter
       })
-    } else {
+    } else if (this.state.autoSplit) {
+      // 3. if we don't have extras and auto-split is on, then fill the texters with assignments
       const factor = 1
       let index = 0
       let skipsByIndex = new Array(newFormValues.texters.length).fill(0)
-      if (newFormValues.texters.length === 1 && this.state.autoSplit) {
+      if (newFormValues.texters.length === 1) {
         const messagedCount = newFormValues.texters[0].assignment.contactsCount - newFormValues.texters[0].assignment.needsMessageCount
         newFormValues.texters[0].assignment.contactsCount = this.formValues().contactsCount
         newFormValues.texters[0].assignment.needsMessageCount = this.formValues().contactsCount - messagedCount
-      } else if (newFormValues.texters.length > 1 && (extra > 0 || (extra < 0 && this.state.autoSplit))) {
```

Comment:
Look like this is removing a case where this code would run with autosplit off and extra > 0, but it's not clear to me reading this code what `extra` is, or how it might end up greater than 0, so I don't know if removing that is a mistake or intentional. Maybe we can walk through how this works?

Reply:
Yes, that's the missed condition I believe is causing the bug.

Ill add a change to rename extra -> unassignedContactCount which I think is more clear and descriptive